app/actioncommands.cpp

```diff
@@ -107,9 +107,37 @@ Status ActionCommands::importSound()
     return st;
 }
 
+void ActionCommands::updateFrameEnd()
+{
+    int layerCount = mEditor->object()->getLayerCount();
+    int projectLength = mEditor->layers()->projectLength();
+    if ( mExportMovieDialog->useKeyFrameLength() )
+    {
+        int maxPosition = 0;
+        Layer* layer;
+        for(int i = 0; i < layerCount; i++)
+        {
+            layer = mEditor->layers()->getLayer( i );
+
+            if ( layer->isLayerPaintable( layer->type() ) )
+            {
+                maxPosition = layer->getMaxKeyFramePosition();
+            }
+        }
+        mExportMovieDialog->setEndFrame( maxPosition );
+    }
+    else
+    {
+        mExportMovieDialog->setEndFrame( projectLength );
+    }
+}
+
```

Comment:
looks like this function should be in ExportMovieDailog, not in ActionCommands.

Reply:
I would suggest 
1. Move this function to layer manager to count project length without sound layers.
2. Pass one more parameter like this `ExportMovieDialog::setDefaultRange( start, end, endPaintable )`
3. Move the checkbox connection into ExportMovieDialog class.

The code structure will be more clear and reasonable.

Reply:
Yeah you may be right about that, alright I'll look into that